src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -673,35 +669,19 @@ private static boolean isTokenWithNoChildSlist(int tokenType) {
         }
 
         /**
-         * Collects validation details for loops' tokens.
+         * Collects validation details for LITERAL_DO loops' tokens.
          *
          * @param ast a {@code DetailAST} value
          * @return an object containing all details to make a validation
          */
-        private static Details getDetailsForLoops(DetailAST ast) {
+        private static Details getDetailsForDoLoops(DetailAST ast) {
+            final DetailAST lcurly = ast.findFirstToken(TokenTypes.SLIST);
+            final DetailAST nextToken = ast.findFirstToken(TokenTypes.DO_WHILE);
             DetailAST rcurly = null;
-            final DetailAST lcurly;
-            final DetailAST nextToken;
-            final int tokenType = ast.getType();
-            final boolean shouldCheckLastRcurly;
-            if (tokenType == TokenTypes.LITERAL_DO) {
```

Comment:
With the cascade of both at https://github.com/checkstyle/checkstyle/pull/12618#discussion_r1066077473 , this always become true and thus all this code can be reduced.